PWGHF/vertexingHF/AliAnaTaskSEXic0SL.C

```diff
@@ -0,0 +1,947 @@
+/*
+   Xic0 -> eXi analysis (* newly written, Aug. 2022)
+
+   Chong Kim
+   Pusan National University
+   kimc@cern.ch
+*/
+
+#include "AliAnaTaskSEXic0SL.h"
+ClassImp(AliAnaTaskSEXic0SL);
+
+#include "AliAnalysisManager.h"
+#include "AliAODcascade.h"
+#include "AliAODMCParticle.h"
+#include "AliAODTrack.h"
+#include "AliESDtrack.h"
+#include "AliESDtrackCuts.h"
+#include "AliESDVertex.h"
+
+#include "AliInputEventHandler.h"
+#include "AliMCEvent.h"
+#include "AliMultSelection.h"
+#include "AliNormalizationCounter.h"
+#include "AliPIDResponse.h"
+#include "AliPPVsMultUtils.h"
+#include "AliRDHFCutsXictoeleXifromAODtracks.h" //For fEvtCuts
+#include "AliVEvent.h"
+#include "AliVVertex.h"
+#include "THistManager.h" //For fHisto
+
+#include "TBranch.h"
+#include "TChain.h"
+#include "TDatabasePDG.h"
+#include "TLeaf.h"
+#include "TString.h"
+#include "TTree.h"
+
+#include <cmath>
+#include <iostream>
+#include <vector>
+using namespace std;
+
+AliAnaTaskSEXic0SL::AliAnaTaskSEXic0SL():AliAnalysisTaskSE("AliAnaTaskSEXic0SL")
+{
+	ControlAnaObjects(0);
+	ControlOutputContainers(0);
+	ResetTreeVariables();
+}//Constructor, default
+
+AliAnaTaskSEXic0SL::AliAnaTaskSEXic0SL(const char* name, const char* option):
+	AliAnalysisTaskSE(name), fTaskOpt(option)
+{
+	ControlAnaObjects(0);
+	ControlOutputContainers(0);
+	ResetTreeVariables();
+}//Constructor
+
+AliAnaTaskSEXic0SL::~AliAnaTaskSEXic0SL()
+{
+	ControlAnaObjects(1);
+}//Destructor
+
+void AliAnaTaskSEXic0SL::Terminate(Option_t *)
+{
+	cout <<"\nDone!\n";
+	return;
+}//Terminate
+
+//=======================================================================================
+void AliAnaTaskSEXic0SL::UserCreateOutputObjects()
+{
+	//Eventwise cut, for MB (default)
+	fEvtCutsMB = new AliRDHFCutsXictoeleXifromAODtracks();
+	fEvtCutsMB->SetOptPileup(AliRDHFCuts::kRejectMVPileupEvent); //Multi vertexer pileup rejection
+	fEvtCutsMB->SetTriggerClass("");
+	fEvtCutsMB->SetTriggerMask(AliVEvent::kINT7);
+	fEvtCutsMB->SetUseInt7TriggerPP2012();
+	fEvtCutsMB->SetUsePhysicsSelection(true);
+
+	//Eventwise cut, for HMV0 trigger
+	fEvtCutsHMV0 = new AliRDHFCutsXictoeleXifromAODtracks();
+	fEvtCutsHMV0->SetOptPileup(AliRDHFCuts::kRejectMVPileupEvent);
+	fEvtCutsHMV0->SetTriggerClass("");
+	fEvtCutsHMV0->SetTriggerMask(AliVEvent::kHighMultV0);
```

Comment:
Done. I created another RDFH object and exclusively (with the existing ones) use it

Reply:
Please write a new reply instead of editing it, so we get a notification. I noticed this by accident now.

1) Ok, fair point. But then please add a check if the event selections in both the MB and HM objects are the same. Something like: [AliAnalysisTaskSEHFTreeCreatorApply.cxx#L756-L761](https://github.com/alisw/AliPhysics/blob/master/PWGHF/vertexingHF/vHFML/AliAnalysisTaskSEHFTreeCreatorApply.cxx#L756-L761). Note you should change the trigger briefly to `kAny` for this check to work

2) Event selections are also important when analysing MC, please add the `AliRDHFCuts::IsEventSelected()` call also for MC. You should use the same objects as you give to data, just changing the trigger to `kAny`

Reply:
1.
The following lines are the current code implementation, temporarily modified for the test you asked for.

```
        //Eventwise cut, for MB (default)
        fEvtCutsMB = new AliRDHFCutsXictoeleXifromAODtracks();
        fEvtCutsMB->SetOptPileup(AliRDHFCuts::kRejectMVPileupEvent); //Multi vertexer pileup rejection
        if (IsMC==false)
        {
            //fEvtCutsMB->SetTriggerMask(AliVEvent::kINT7);
            //fEvtCutsMB->SetUseInt7TriggerPP2012();
            fEvtCutsMB->SetTriggerMask(AliVEvent::kAny);
        }
        else if (IsLegacy==false) fEvtCutsMB->SetTriggerMask(AliVEvent::kAny);
        fEvtCutsMB->SetUsePhysicsSelection(true);

        //Eventwise cut, for HMV0 trigger
        fEvtCutsHMV0 = new AliRDHFCutsXictoeleXifromAODtracks();
        fEvtCutsHMV0->SetOptPileup(AliRDHFCuts::kRejectMVPileupEvent);
        if (IsMC==false)
        {
            //fEvtCutsHMV0->SetTriggerMask(AliVEvent::kHighMultV0);
            fEvtCutsHMV0->SetTriggerMask(AliVEvent::kAny);
        }
        else if (IsLegacy==false) fEvtCutsHMV0->SetTriggerMask(AliVEvent::kAny);
        fEvtCutsHMV0->SetUsePhysicsSelection(true);
```
And then, I tested if the response is the same by using a sample AOD, as follows.
```
        //Activate RDHF cut by invoking it
        fEvtGoodMB   = fEvtCutsMB  ->IsEventSelected(fEvt);
        fEvtGoodHMV0 = fEvtCutsHMV0->IsEventSelected(fEvt);

        cout <<Form("EvtID: %5i, IsEvtSelected - MB/HMV0: %i/%i %s\n",
                fEvtID, fEvtGoodMB, fEvtGoodHMV0, (fEvtGoodMB==fEvtGoodHMV0)?"":"!@#");
```
The results are exactly the same. I'll send the log if you need to check.

2.